File: conference/registration-bc/web/src/main/java/com/microsoft/conference/registration/domain/Orders/DefaultPricingService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using ECommon.Components;

namespace Registration.Orders
{
    [Component]
    public class DefaultPricingService : IPricingService
    {
        public OrderTotal CalculateTotal(Guid conferenceId, IEnumerable<SeatQuantity> seatQuantityList)
        {
            var orderLines = new List<OrderLine>();
            foreach (var seatQuantity in seatQuantityList)
            {
                orderLines.Add(new OrderLine(seatQuantity, Math.Round(seatQuantity.Seat.UnitPrice * seatQuantity.Quantity, 2)));
            }
            return new OrderTotal(orderLines.ToArray(), orderLines.Sum(x => x.LineTotal));
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/Orders/DefaultPricingService.java
package com.microsoft.conference.registration.domain.Orders;

import com.microsoft.conference.registration.domain.Orders.Models.OrderLine;
import com.microsoft.conference.registration.domain.Orders.Models.OrderTotal;
import com.microsoft.conference.registration.domain.SeatQuantity;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

public class DefaultPricingService implements IPricingService {
    @Override
    public OrderTotal CalculateTotal(String conferenceId, List<SeatQuantity> seatQuantityList) {
        List<OrderLine> orderLines = new ArrayList<>();
        for (SeatQuantity seatQuantity : seatQuantityList) {
            orderLines.add(new OrderLine(seatQuantity, seatQuantity.Seat.UnitPrice.multiply(BigDecimal.valueOf(seatQuantity.Quantity))));
        }
        BigDecimal total = BigDecimal.ZERO;
        for (OrderLine x : orderLines) {
            total = total.add(x.LineTotal);
        }
        return new OrderTotal(orderLines, total);
    }
}
